src/Classes/PartyTab.lua

```diff
@@ -0,0 +1,620 @@
+				if self.controls.importCodeMode.selIndex == 1 or self.controls.importCodeMode.selIndex == 2 then
+					if #self.controls.editAuras.buf > 0 then
+						node[5].attrib.string = self.controls.editAuras.buf.."\n"..node[5].attrib.string
+					end
+					self.controls.editAuras:SetText(node[5].attrib.string)
+					self:ParseBuffs(self.processedInput["Aura"], self.controls.editAuras.buf, "Aura")
+				end
+				if self.controls.importCodeMode.selIndex == 1 or self.controls.importCodeMode.selIndex == 3 then
+					if #self.controls.editCurses.buf > 0 then
+						node[6].attrib.string = self.controls.editCurses.buf.."\n"..node[6].attrib.string
+					end
+					self.controls.editCurses:SetText(node[6].attrib.string)
+					self:ParseBuffs(self.processedInput["Curse"], self.controls.editCurses.buf, "Curse")
+				end
+				if self.controls.importCodeMode.selIndex == 1 or self.controls.importCodeMode.selIndex == 4 then
+					if #self.controls.enemyCond.buf > 0 then
+						node[7].attrib.string = self.controls.enemyCond.buf.."\n"..node[7].attrib.string
+					end
+					self.controls.enemyCond:SetText(node[7].attrib.string)
+				end
+				if self.controls.importCodeMode.selIndex == 1 or self.controls.importCodeMode.selIndex == 5 then
+					if #self.controls.enemyMods.buf > 0 then
+						node[8].attrib.string = self.controls.enemyMods.buf.."\n"..node[8].attrib.string
+					end
+					self.controls.enemyMods:SetText(node[8].attrib.string)
+				end
+				if self.controls.importCodeMode.selIndex == 1 or self.controls.importCodeMode.selIndex == 4 or self.controls.importCodeMode.selIndex == 5 then
+					wipeTable(self.enemyModList)
+					self.enemyModList = new("ModList")
+					self:ParseBuffs(self.enemyModList, self.controls.enemyCond.buf, "EnemyConditions")
+					self:ParseBuffs(self.enemyModList, self.controls.enemyMods.buf, "EnemyMods")
+				end
+				self.build.buildFlag = true 
+				break
+			end
+		end
+	end
+	
+	self.controls.importCodeIn = new("EditControl", {"TOPLEFT",self.controls.importCodeHeader,"BOTTOMLEFT"}, 0, 4, 328, 20, "", nil, nil, nil, importCodeHandle)
+	self.controls.importCodeIn.width = function()
+		return (self.width > 880) and 328 or (self.width / 2 - 100)
+	end
+	self.controls.importCodeIn.enterFunc = function()
+		if self.importCodeValid then
+			self.controls.importCodeGo.onClick()
+		end
+	end
+	self.controls.importCodeState = new("LabelControl", {"LEFT",self.controls.importCodeIn,"RIGHT"}, 8, 0, 0, 16)
+	self.controls.importCodeState.label = function()
+		return self.importCodeDetail or ""
+	end
+	self.controls.importCodeMode = new("DropDownControl", {"TOPLEFT",self.controls.importCodeIn,"BOTTOMLEFT"}, 0, 4, 160, 20, { "All", "Aura", "Curse", "EnemyConditions", "EnemyMods" })
+	self.controls.importCodeMode.enabled = function()
+		return self.importCodeValid
+	end
+	self.controls.importCodeMode2 = new("DropDownControl", {"LEFT",self.controls.importCodeMode,"RIGHT"}, 8, 0, 160, 20, { "Replace", "Append", "Clear" })
```

Comment:
Can we rename these so it's a little more clear what is happening?  e.g. importCodeDestSelect and editModeSelect or something